website/project/model.py

```diff
@@ -3032,3 +3031,72 @@ def approve_embargo(self, user, token):
                 },
                 auth=Auth(user),
             )
+
+class DraftRegistration(AddonModelMixin, StoredObject):
+
+    is_draft_registration = True
+
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    datetime_initiated = fields.DateTimeField(auto_now_add=True)
+    datetime_updated = fields.DateTimeField(auto_now=True)
+
+    branched_from = fields.ForeignField('node')
+
+    initiator = fields.ForeignField('user')
+
+    registration_metadata = fields.DictionaryField({})
+    registration_schema = fields.ForeignField('metaschema')
+    registered_node = fields.ForeignField('node')
+
+    is_pending_review = fields.BooleanField(default=False)
+
+    schema_name = fields.StringField()
+
+    storage = fields.ForeignField('osfstoragenodesettings')
+
+    # proxy fields from branched_from Node
+    def __getattr__(self, attr):
+        try:
+            return self.__dict__[attr]
+        except KeyError:
+            return getattr(self.branched_from, attr, None)
+
+    def get_comments(self):
+        """ Returns a list of all comments made on a draft in the format of :
+        [{
+            'page1': [{
+                user: 'user',
+                last_modified: 'Thu, 09 Jul 2015 21:45:56 GMT',
+                value: 'Comment text'
+            }],
+            'page2': [{
+                user: 'user',
+                last_modified: 'Thu, 09 Jul 2015 21:45:56 GMT',
+                value: 'Comment text'
+            },
+            {
+                user: 'user',
+                last_modified: 'Thu, 09 Jul 2015 21:45:56 GMT',
+                value: 'Comment text'
+            }]
+        }]
+        """
+        pages = self.registration_schema['schema']['pages']
+        all_comments = list()
+
+        # there has to be a more pythonic way to do this
+        page_num = 1
+        for page in pages:
+            comment_obj = dict()
+            comments = page['comments']
+
+            comment_obj['page{}'.format(page_num)] = list()
```

Comment:
Ah, this was a mistake made by looking at `registration_schema` instead of `registration_metadata`. Will fix